src/sip-dialog-controller.cpp

```diff
@@ -1677,7 +1677,8 @@ namespace drachtio {
         std::shared_ptr<RIP> rip  ;
 
         nta_leg_t* leg = nta_leg_by_call_id(m_pController->getAgent(), sip->sip_call_id->i_id);
-        assert(leg) ;
+        if (!leg)
+          return 0;
```

Comment:
could we have a log message here to indicate what has happened - ie the possibility of glare condition

Reply:
Sure

Reply:
Done